main/src/io/file_utils.hpp

```diff
@@ -47,7 +48,9 @@ void writeAscii(size_t firstIndex, size_t lastIndex, const std::string& path, bo
             for (auto field : fields)
             {
                 [[maybe_unused]] std::initializer_list<int> list{(dumpFile << separators, 0)...};
-                dumpFile << field[i];
+                std::visit([&dumpFile, i](auto& arg) {
+                    dumpFile << arg[i];
+                }, field);
```

Comment:
You should be able to pass `field[i]` directly here and avoid capture-by-value of `i`.

Reply:
I tried to do that too but, as field is a variant, I am not able to dereference it directly there. Within the visit function when the type is known we can dereference. The compiler complains when I try to pass field[i].

Reply:
ah yes, correct.